### DISCUSSION TITLE
in action, update state values, but subscribe is just run once

### DISCUSSION THREAD

**Discussion by seho-dev** [vuejs/pinia#1421] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/vitejs-vite-racwjf?file=package.json,src%2Fstore%2Fuser.ts,src%2Fmain.ts,src%2Fcomponents%2FHelloWorld.vue&terminal=dev

### Steps to reproduce the bug

https://stackblitz.com/edit/vitejs-vite-racwjf?file=package.json,src%2Fstore%2Fuser.ts,src%2Fmain.ts,src%2Fcomponents%2FHelloWorld.vue&terminal=dev

### Expected behavior

https://stackblitz.com/edit/vitejs-vite-racwjf?file=package.json,src%2Fstore%2Fuser.ts,src%2Fmain.ts,src%2Fcomponents%2FHelloWorld.vue&terminal=dev

### Actual behavior

https://stackblitz.com/edit/vitejs-vite-racwjf?file=package.json,src%2Fstore%2Fuser.ts,src%2Fmain.ts,src%2Fcomponents%2FHelloWorld.vue&terminal=dev

**Comment by seho-dev** [comment]
What is the configuration of flush, what does it mean, I solved my problem with flush

**Comment by posva** [comment, marked as answer]
You didn't say what you were expecting and what to do. The action is called before the subscribe so it makes sense to not trigger the subscribe.

About flush: https://vuejs.org/guide/essentials/watchers.html#callback-flush-timing

**Reply by seho-dev** [reply]
But why is it only triggered once?

**Reply by posva** [reply]
because flush allows watchers to group changes and `$subscribe()` takes does the same
